### DISCUSSION TITLE
Unable To Insert Into Table From Client After Creating Trigger

### DISCUSSION THREAD

**Discussion by jason-dubon** [supabase/supabase#18916] [Questions] [answered]
Hello everyone, 

I am using Supabase in my swift project.

I am running into an issue where after I create a trigger for an insert that happens to a table. Then, I am unable to insert to that table from the client side. When I delete the trigger, the insert works perfectly fine. 

I tried changing my function security to Definer and that still gave me issues. 

Could this be an issue on the Supabase side? or is it caused by the Supabase Swift SDK ? or am I messing up somewhere 🤔

**Comment by GaryAustin1** [comment]
You are not giving much to go on.  Are there in errors in the Postgres logs?
Not likely Supabase related as triggers are a Postgres thing.
Please show your trigger code at least if the log does not tell you what the error is.

**Reply by jason-dubon** [reply]
This is the trigger created in supabase
![Screenshot 2023-11-13 at 5 53 00 PM](https://github.com/supabase/supabase/assets/15134835/fedb388a-dbc9-46e5-bd18-1a8d2ab8e6f1)


This is the function that was created. 
![Screenshot 2023-11-13 at 5 52 10 PM](https://github.com/supabase/supabase/assets/15134835/2ffdb0a4-cade-4688-b730-9497b4cb0cf2)

**Reply by jason-dubon** [reply, marked as answer]
Well now I feel dumb, i was able to resolve the issue. I was missing a column in my messages table that would allow me to reference the channel_uid

**Reply by jason-dubon** [reply]
Thanks!
